packages/xlayers/src/app/editor/code-editor/editor-container/codegen/shared-codegen.service.ts

```diff
@@ -10,7 +11,10 @@ export enum Template {
 })
 export class SharedCodegen {
   private indentationSymbol = '  '; // 2 spaces ftw
-
+  generateComponentOptimizedStyles(ast: SketchMSLayer) {
+    const optimizer = new StyleSheetOptimizer(true);
```

Comment:
we should have this as an angular service.

Reply:
I was thinking about that do we really want this. For now I will make the change we can always refactor it.

Reply:
this would allow people to easily switch implementation if they like.